data/libs/Ship.lua

```diff
@@ -51,260 +58,23 @@ local CrewRoster = {}
--- Status:
---
---   experimental
---
-function Ship:AddEquip(item, count, slot)
-	local ret = self.equipSet:Add(self, item, count, slot)
-	if ret > 0 then
-		Event.Queue("onShipEquipmentChanged", self, item)
-	end
-	return ret
-end
-
---
--- Method: GetEquip
---
--- Get a list of equipment in a given equipment slot
---
--- > equip = ship:GetEquip(slot, index)
--- > equiplist = ship:GetEquip(slot)
---
--- Parameters:
---
---   slot  - a slot name string (e.g., "autopilot")
---
---   index - optional. The equipment position in the slot to fetch. If
---           specified the item at that position in the slot will be returned,
---           otherwise a table containing all items in the slot will be
---           returned instead.
---
--- Return:
---
---   equip - when index is specified, an equipment type object for the
---           item, or nil
---
---   equiplist - when index is not specified, a table which has slot indexes
---               as keys and equipment type objects as values.
---               WARNING: although slot indexes are integers, this table is
---               not guaranteed to contain a contiguous set of entries, so you
---               should iterate over it with pairs(), not ipairs().
---
--- Availability:
---
---  alpha 10
---
--- Status:
---
---  experimental
---
-Ship.GetEquip = function (self, slot, index)
-	return self.equipSet:Get(slot, index)
-end
-
---
--- Method: GetEquipFree
---
--- Get the amount of free space in a given equipment slot
---
--- > free = ship:GetEquipFree(slot)
---
--- Parameters:
---
---   slot - a slot name (e.g., "autopilot")
---
--- Return:
---
---   free - the number of item spaces left in this slot
---
--- Availability:
---
---  alpha 10
---
--- Status:
---
---  experimental
---
-Ship.GetEquipFree = function (self, slot)
-	return self.equipSet:FreeSpace(slot)
-end
-
---
--- Method: SetEquip
---
--- Overwrite a single item of equipment in a given equipment slot
---
--- > ship:SetEquip(slot, index, equip)
---
--- Parameters:
---
---   slot - a slot name (e.g., "laser_front")
---
---   index - the position to store the item in
---
---   item - an equipment type object (e.g., Equipment.laser.large_plasma_accelerator)
---
--- Example:
+-- Method: GetInstalledHyperdrive
 --
--- > -- add a laser to the rear laser mount
--- > ship:SetEquip("laser_rear", 1, Equipment.laser.pulsecannon_4mw)
+-- Return the ship's installed hyperdrive equipment item if present.
 --
--- Availability:
---
---  alpha 10
+-- > if ship:GetInstalledHyperdrive() then HyperdriveOverloadAndExplode(ship) end
 --
 -- Status:
 --
---  experimental
+--   stable
 --
-Ship.SetEquip = function (self, slot, index, item)
-	self.equipSet:Set(self, slot, index, item)
-	Event.Queue("onShipEquipmentChanged", self)
+---@return Equipment.HyperdriveType? hyperdrive
+function Ship:GetInstalledHyperdrive()
+	---@type Equipment.HyperdriveType[]
+	local drives = self:GetComponent("EquipSet"):GetInstalledOfType("hyperdrive")
+	return drives[1]
```

Comment:
Other code mentions that there can only be a single hyperdrive in a ship. Should we perform a sanity check here?

Reply:
This is a commonly-used accessor, so I'd prefer not to add extra runtime overhead if possible. More than one hyperdrive would be the fault of the HullConfig's slot layout violating one of the gameplay/lore constraints of the game, which I'd prefer to catch at a different, higher level (i.e. at startup) if at all.